edge/pkg/eventbus/mqtt/filter.go

```diff
@@ -0,0 +1,158 @@
+package mqtt
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+type HandlerFunc func(topic string, payload []byte)
+
+// MessageMuxEntry message mux entry
+type MessageMuxEntry struct {
+	pattern     *MessagePattern
+	handlerFunc HandlerFunc
+}
+
+// MessagePattern message pattern
+type MessagePattern struct {
+	resource  string
+	operation string
+
+	resExpr *MessageExpression
```

Comment:
hi @Shelley-BaoYue, move the value of the pointer type to the first line of the struct?

Reply:
> hi @Shelley-BaoYue, move the value of the pointer type to the first line of the struct?

done